### PR TITLE
feat: Show number of pod restarts in the argo ui (5041)

### LINKED ISSUE #5041: Show number of pod restarts in the argo ui
Show number of pod restarts in the argo ui.

# Motivation
 Showing number of pod restart counts in the argo ui will help us to see which pods were restarted and how many times without going to any other monitoring dashboards or running kubectl commands. This will be very helpful feature. 

# Proposal

By getting the number of restart count, we can add this count at the bottom of each pod as shown by arrow in the screenshot
image

### PR DESCRIPTION
**Pull request by keithchong**
Signed-off-by: Keith Chong <kykchong@redhat.com>

Fixes https://github.com/argoproj/argo-cd/issues/5041

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Review by rbreeze — COMMENTED** [review]
Thanks Keith, LGTM! Will wait for @alexmt 's approval to merge.

**Review by alexmt — APPROVED** [review]
LGTM

**Pull request merged**
